refinery/tool_manager/models.py

```diff
@@ -237,38 +243,88 @@ def __str__(self):
     def launch(self):
         if self.get_tool_type() == ToolDefinition.VISUALIZATION:
             container = DockerContainerSpec(
-                image_name=self.tool_definition.docker_image_name,
+                image_name=self.tool_definition.image_name,
                 container_name=self.container_name,
                 labels={self.uuid: ToolDefinition.VISUALIZATION},
                 container_input_path=(
                     self.tool_definition.container_input_path
                 ),
-                input={"file_relationships": self.file_relationships}
+                input={"file_relationships": eval(self.file_relationships)},
+                manager=self.get_django_docker_engine_manager()
             )
             try:
                 container.run()
             except APIError as e:
                 return HttpResponseServerError(content=e)
             else:
-                return HttpResponse(self.get_relative_container_url())
+                return JsonResponse(
+                    {
+                        "tool_url": self.get_relative_container_url()
+                    }
+                )
 
         if self.get_tool_type() == ToolDefinition.WORKFLOW:
             raise NotImplementedError
 
+    def get_django_docker_engine_manager(self):
+        """
+        Helper method to return the proper managerial class for
+        django_docker_engine
+        """
+        if "aws" in os.getenv("DJANGO_SETTINGS_MODULE"):
+            raise NotImplementedError
+        else:
+            return LocalManager()
+
     def get_relative_container_url(self):
         """
         Construct & return the relative url of our Tool's container
         """
-        return "/visualizations/{}".format(self.container_name)
+        return "/{}/{}".format(
+            settings.DJANGO_DOCKER_ENGINE_URL_ROOT,
+            self.container_name
+        )
 
     def get_tool_name(self):
         return self.tool_definition.name
 
     def get_tool_type(self):
         return self.tool_definition.tool_type
 
-    def parse_file_relationships_string(self):
-        raise NotImplementedError
+    def update_file_relationships_string(self):
+        """
+        Replace a Tool's Node uuids in its `file_relationships` string with
+        their respective FileStoreItem's urls. No error handling here since
+        this method is only called in an atomic transaction.
+        """
+        populated_file_relationships = re.sub(
+            r'[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}',
+            self.populate_url_from_node_uuid,
+            self.file_relationships
+        )
+        self.file_relationships = populated_file_relationships
+        self.save()
 
-    def populate_url_from_node_uuid(self):
-        raise NotImplementedError
+    @staticmethod
+    def populate_url_from_node_uuid(matched_object):
```

Comment:
re.sub is providing me with a matched object instance in this case,  I think it would make more sense to rename the method rather than the argument to then still have to say:
`node = Node.objects.get(uuid=uuid.group(0))`